buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts

```diff
@@ -5,7 +5,7 @@ plugins {
     id("signing")
 }
 
-val isARelease = project.hasProperty("release") && project.property("release") == "true"
+val isARelease = System.getenv("CI") != null
```

Comment:
@trask I noticed that this env var check is what is being used [in here](https://github.com/open-telemetry/opentelemetry-java/blob/96c895893c8fb8aeb7e786a050e772d087513f01/buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts#L60) to tell when the compilation is running in a GH action vs anywhere else, which is really cool and cleaner than having to pass a flag to the command, so we no longer need the `release` param 👍